src/internal/m365/onedrive/restore.go

```diff
@@ -747,6 +743,50 @@ func createRestoreFolders(
 	return parentFolderID, nil
 }
 
+func createFolder(
+	ctx context.Context,
+	piic PostItemInContainerer,
+	driveID, parentFolderID, folderName string,
+) (models.DriveItemable, error) {
+	// create the folder if not found
+	// the Replace collision policy is used since collisions on that
+	// policy will no-op and return the existing folder.  This has two
```

Comment:
given the amount of churn we've seen on the graph API I kind of want to say we should have a regression test checking specifically that attempting to (re)create an existing folder with items with the replace policy doesn't disturb the items already in the folder. Or if there's explicit documentation of this fact drop a link to it here so we can know the behavior going forward

Reply:
There's no explicit documentation, but from research into the condition a replace conflict is equivalent to a GET request.  I'm trying to parse your request, and am not sure if you're suggesting we cannot trust the server-side replace behavior, or if we need a regression test to ensure we don't accidentally change things on our side.  If the former, I feel like that's stretching the value a little thin.  The latter sounds very useful, though.  Going to add a regression test either way, but am curious about the clarification here.

Reply:
I want to see a test that ensures graph behaves the way we expect it to. I want the test because otherwise if the behavior of graph does change we'll never know we're disappearing items until someone comes and tells us about it, which could take a very long time

For example, imagine the behavior of graph changed such that replace on a folder would delete the original folder (and all items/subfolders in it) and create a new folder. If someone had made a backup from a while back and then tried to do an in-place restore with it we could end up deleting items they may not have a backup for

At least if we have a test we'll know when stuff breaks so we can mitigate it. If it were documented behavior I'd be more open to just having a link to it and pointing to that. However, since it's not documented we need to know what the actual behavior will be in-practice and we need to keep checking the behavior continues to be what we expect

Reply:
Yeah, unfortunately docs are less that skeletal when it comes to conflict behavior.  Not much we can reference there for reassurance.

Reply:
@ashmrtn regression test added.  This is ready for another review, please.